open-machine-learning-jupyter-book/ml-fundamentals/ml-overview.md

```diff
@@ -0,0 +1,616 @@
+
+There is a tradeoff between precision and recall. Often, increasing precision will decrease recall and vice versa. To simplify things, we combine both of these two metrics into a single metric called the *F1 score*. 
+
+F1 score is the harmonic mean of precision and recall, and it shows how good the model is at classifying all classes without having to balance between precision and recall. If either precision or recall is very low, the F1 score is going to be low too.
+
+Both accuracy, precision, and recall can be calculated easily by using a [confusion matrix](https://jeande.tech/how-to-read-a-confusion-matrix). A confusion matrix shows the number of correct and incorrect predictions made by a classifier in all available classes. 
+
+More intuitively, a confusion matrix is made of 4 main elements: True negatives, false negatives, true positives, and false positives.
+
+* **True Positives(TP)**: Number of samples that are correctly classified as positive, and their actual label is positive.
+
+* **False Positives (FP)**: Number of samples that are incorrectly classified as positive, when in fact their actual label is negative.
+
+* **True Negatives (TN)**: Number of samples that are correctly classified as negative, and their actual label is negative.
+
+* **False Negatives (FN)**: Number of samples that are incorrectly classified as negative, when in fact their actual label is positive.
+
+The accuracy that we talked about is the number of correct examples over total examples. So, that is 
+
+*Accuracy = (TP + TN) / (TP + TN + FP + FN)*
+
+Precision is the model accuracy on predicting positive examples. 
+
+*Precision = TP / (TP + FP)*
+
+On the other hand, Recall is the model ability to predict the positive examples correctly. 
+
+*Recall = TP / (TP+FN)*
+
+The higher the recall and precision, the better the model is at making accurate predictions but there is a tradeoff between them. Increasing precision will reduce the recall and vice versa. 
+
+A classifier that doesn't have false positives has a precision of 1, and a classifier that doesn't have false negatives has a recall of 1. Ideally, a perfect classifier will have the precision and recall of 1.
+
+We can combine both precision and recall to get another metric called F1 Score. F1 Score is the harmonic mean of precision and recall. 
+
+*F1 Score = 2 x(precision x recall) / (precision + recall)*
+
+Take an example of the following confusion matrix. 
+
+```{figure} ../../images/ml-fundamentals/ml-overview/confusion-matrix.png
+---
+name: 'confusion-matrix.png'
+width: 90%
+---
+An example of confusion matrix
+```
+
+
+From the above confusion matrix:
+
+* `Accuracy = (TP + TN) / (TP + TN + FP + FN) = (71 +36)/(71+36+7+0) = 0.93 or 93%`
+* `Precision = TP / (TP + FP) = 71/(71+7) =0.91 or  91%`
+* `Recall = TP / (TP + FN) = 71/(71+0) = 1, or 100%`
+* `F1 score = 2PR / (P + R) = 2x0.91x1/(0.91+1) = 0.95, or 95%`
```

Comment:
These are not code, instead, they are formulation. We should use latex? E.g. $(TP + TN) / (TP + TN + FP + FN) = (71 +36)/(71+36+7+0) = 0.93 or 93%$

Reply:
This section is now removed.